src/main/java/sirius/biz/packages/Packages.java

```diff
@@ -0,0 +1,119 @@
+import sirius.kernel.di.std.Register;
+import sirius.kernel.nls.NLS;
+import sirius.kernel.settings.Extension;
+import sirius.web.security.UserContext;
+import sirius.web.security.UserInfo;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * This helper class is a wrapper for the <tt>security.packages</tt> configuratuions.
+ * <p>
+ * For a highlevel overview see the README.md of this package.
+ */
+@Register(classes = Packages.class)
+public class Packages {
+
+    /**
+     * Returns the list of packages available for the given scope.
+     *
+     * @param scope the scope
+     * @return the pricePackages of the scope
+     */
+    public List<String> getPackages(String scope) {
+        return getScopeExtension(scope).getStringList("packages");
+    }
+
+    private Extension getScopeExtension(String scope) {
+        return Sirius.getSettings().getExtension("security.packages", scope);
+    }
+
+    /**
+     * Returns the translated name for the given package of the given scope.
+     *
+     * @param scope      the scope
+     * @param packageKey the package key
+     * @return the translated name for the given package of the given scope
+     */
+    public String getPackageName(String scope, String packageKey) {
+        return NLS.get("Packages.package." + scope + "." + packageKey);
+    }
+
+    /**
+     * Returns the list of upgrades available for the given scope.
+     *
+     * @param scope the scope
+     * @return the upgrades of the scope
+     */
+    public List<String> getUpgrades(String scope) {
+        return getScopeExtension(scope).getStringList("upgrades");
+    }
+
+    /**
+     * Returns the translated name for the given upgrade of the given scope.
+     *
+     * @param scope      the scope
+     * @param upgradeKey the upgrade key
+     * @return the translated name for the given upgrade of the given scope
+     */
+    public String getUpgradeName(String scope, String upgradeKey) {
+        return NLS.get("Packages.upgrade." + scope + "." + upgradeKey);
+    }
+
+    /**
+     * Checks whether the current user has the required permission for a given permission.
+     *
+     * @param permission the permission in question
+     * @return true if the current user has the required permission, false if not
+     */
+    public boolean hasRequiredPermissionForPermission(String permission) {
+        UserInfo currentUser = UserContext.get().getUser();
+        return hasRequiredPermissionForPermission(permission, currentUser::hasPermission);
+    }
+
+    /**
+     * Checks whether the permissionData object has the required permission for a given permission.
+     *
+     * @param permission     the permission in question
+     * @param permissionData the permissionData object to check
+     * @return true if the permissionData object has the required permission, false if not
+     */
+    public boolean hasRequiredPermissionForPermission(String permission, PermissionData permissionData) {
```

Comment:
rather use a Function<String, Boolean>

and also maybe transfer the permission logic from UserInfo to here and apply that before. The check features this will most probably not be called with either a user nor with a PermissionData object but rather a set of features